Content.Shared/Movement/Systems/SlowContactsSystem.cs

```diff
@@ -60,18 +35,29 @@ private void OnHandleState(EntityUid uid, SlowContactsComponent component, ref C
         component.SprintSpeedModifier = state.SprintSpeedModifier;
     }
 
-    private void MovementSpeedCheck(EntityUid uid, SlowedByContactComponent component, RefreshMovementSpeedModifiersEvent args)
+    public override void UpdateBeforeSolve(bool prediction, float frameTime)
     {
-        if (!EntityManager.TryGetComponent<PhysicsComponent>(uid, out var physicsComponent))
-            return;
+        base.UpdateBeforeSolve(prediction, frameTime);
 
+        var query = EntityQueryEnumerator<SlowedByContactComponent, MovementSpeedModifierComponent>();
+
+        // TODO: I really hate this but I have tried for an hour to fix the mispredict yet somehow
+        // it still felt worse than conveyors.
+        // It may just be the contact mispredict but that requires significant physics work + optimisation +
+        // box2d 3.0 is coming so all of it may get uplifted.
+        while (query.MoveNext(out var uid, out _, out var modifier))
+        {
+            _speedModifierSystem.RefreshMovementSpeedModifiers(uid, modifier);
+        }
+    }
```

Comment:
I'm still not entirely happy about this but doing proper contact prediction will require several thousand lines of physics code and a big refactor is coming anyway.